### ISSUE TITLE
bug: ion-datetime date and time pickers do not render correctly on (some) devices with maximum font size set

### ISSUE THREAD

**Issue by ppolce** [ionic-team/ionic-framework#25964]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

Date and Time pickers do not render correctly on Samsung Galaxy S21 when the font size is set to largest. 
Labels overlap and make the text unreadable.

NOTE: It only seem to affect **some** android phones (e.g. Galaxy S21).

### Expected Behavior

Rows should not overlap. Text should be visible at any font size setting.

### Steps to Reproduce

1. ionic start choose Angular / Tabs application
2. edit tab1.page.html, comment out (or remove) the `<app-explorer-container>` line
3. add `<ion-datetime></ion-datetime>`
4. add android `npm i @capacitor/android`
5. build a testapp.apk
6. Install the apk on an android device that allows for very large fonts (in my case, Samsung **Galaxy S21**).
7. Open settings and choose the **largest** font size available

Run the app. A page with the current month's calendar appears. 
Click on the date picker, you’ll see that the months overlap:

![image](https://user-images.githubusercontent.com/3269489/190904308-85abcd73-1768-480e-b3ca-8568f46f7c55.png)

and

![image](https://user-images.githubusercontent.com/3269489/190904423-40625e59-de8e-41cf-88d9-3c020e3e5658.png)

### Ionic Info

```

Ionic:

   Ionic CLI                     : 6.20.1 (/home/develop/.local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.9.4
   @angular-devkit/build-angular : 13.3.9
   @angular-devkit/schematics    : 13.3.9
   @angular/cli                  : 13.3.9
   @ionic/angular-toolkit        : 6.1.0

Capacitor:

   Capacitor CLI      : 4.2.0
   @capacitor/android : 4.2.0
   @capacitor/core    : 4.2.0
   @capacitor/ios     : 4.2.0

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.0

System:

   NodeJS : v14.20.0 (/usr/bin/node)
   npm    : 7.24.2
   OS     : Linux 5.15
```

### Additional Information

same as reported here: https://forum.ionicframework.com/t/ion-datetime-date-and-time-pickers-do-not-render-correctly-on-devices-with-high-accessibility-settings/225750

**Comment by ppolce** [comment]
Hi @liamdebeasi,

I have created a little app that's just the ionic "blank app" with a ion-datetime component on the homepage.
You can find it here: https://github.com/ppolce/ion-datetime-bug

Thanks a lot for your help!
P

**Comment by liamdebeasi** [comment]
Here is a dev build with a proposed fix if you are interested in testing it:

```
npm install @ionic/angular@6.2.8-dev.11664221659.1dd1c3b9
```

**Comment by ppolce** [comment]
Hi,

Tested it. It works as expected.
Thanks a lot!
P

**Comment by Pablito203** [comment]
Hi @liamdebeasi, 

Will this fix be merged into newer versions?

**Comment by luqasmateus** [comment]
Hello @liamdebeasi 

I can't update my apps to the latest @ionic/angular version because of this bug. Can you take a look at this again, please?
The dev build you provided works correctly.

**Comment by Sampath-Lokuge** [comment]
Any reason why this has not been merged yet?

**Comment by liamdebeasi** [comment]
The proposed patch regressed selection behavior, so we are unable to merge the fix in its current state. We'll need to investigate an alternative fix in the future.

**Comment by Sampath-Lokuge** [comment]
Hmm...

Do you know how to solve this issue then? i.e. I cannot see full dates here. Thanks!

This didn't help me:

 <ion-popover
      trigger="end-date-picker"
      show-backdrop="false"
      side="top"
      alignment="start"
    >

![calender](https://user-images.githubusercontent.com/5206526/233662333-e7cd9840-f140-497c-b95a-3fdaef5cf14e.png)

**Comment by liamdebeasi** [comment]
See https://github.com/ionic-team/ionic-framework/issues/24780

**Comment by Sampath-Lokuge** [comment]
Yes, it is disappointing. If you can provide the CSS changes like paddings and etc within calendar dates then we can acquire more vertical spaces within the calendar. Now it has a lot of paddings in between date rows. But we cannot change that due to the shadow dom effect.

**Comment by DevForAll-Loic** [comment]
> The proposed patch regressed selection behavior, so we are unable to merge the fix in its current state. We'll need to investigate an alternative fix in the future.

Hey, my company is asking if there is any news ?
